File: implementations/csharp/src/Skight.AgentPlatform/AutomationTools.cs
using System;
using System.Text.Json;
using System.Threading.Tasks;

namespace Skight.AgentPlatform
{
    public static class AutomationTools
    {
        public static async Task<string> ScheduleTimerAsync(string argsJson)
        {
            try
            {
                using var doc = JsonDocument.Parse(argsJson);
                var root = doc.RootElement;
                if (!root.TryGetProperty("seconds", out var secProp) || !root.TryGetProperty("prompt", out var promptProp))
                {
                    return "Error: Missing 'seconds' or 'prompt' argument.";
                }

                int seconds = secProp.GetInt32();
                string prompt = promptProp.GetString()!;
                string taskId = Guid.NewGuid().ToString("N").Substring(0, 8);

                _ = Task.Run(async () =>
                {
                    await Task.Delay(seconds * 1000);
                    Console.WriteLine($"\n  [Automation Timer Fired] Task #{taskId}: '{prompt}'");
                });

                return $"Timer task #{taskId} scheduled for {seconds} seconds. Prompt: '{prompt}'";
            }
            catch (Exception ex)
            {
                return $"Error scheduling timer: {ex.Message}";
            }
        }
    }
}

File: implementations/fsharp/src/Skight.AgentPlatform.FSharp/AutomationTools.fs
namespace Skight.AgentPlatform.FSharp

open System
open System.Text.Json
open System.Threading.Tasks

module AutomationTools =

    let scheduleTimerAsync (argsJson: string) : Task<string> =
        Task.FromResult(
            try
                let doc = JsonDocument.Parse argsJson
                use _ = doc
                let root = doc.RootElement
                match root.TryGetProperty("seconds"), root.TryGetProperty("prompt") with
                | (true, secProp), (true, promptProp) ->
                    let seconds = secProp.GetInt32()
                    let prompt = promptProp.GetString()
                    let taskId = Guid.NewGuid().ToString("N").Substring(0, 8)

                    Task.Run(fun () ->
                        async {
                            do! Async.Sleep (seconds * 1000)
                            printfn "\n  [Automation Timer Fired] Task #%s: '%s'" taskId prompt
                        } |> Async.StartAsTask :> Task
                    ) |> ignore

                    sprintf "Timer task #%s scheduled for %d seconds. Prompt: '%s'" taskId seconds prompt
                | _ -> "Error: Missing 'seconds' or 'prompt' argument."
            with ex -> sprintf "Error scheduling timer: %s" ex.Message
        )
